### ISSUE TITLE
get user by id do not search in custom tenant with custom database

### ISSUE THREAD

**Issue by edubacco** [supertokens/supertokens-core#941]
## 🐛 Bug Report

Hi folks,

in my hosted supertokens instances with nodejs backend and multitenancy active, it seems that the function https://supertokens.com/docs/nodejs/modules/index.html#getUser
only looks for user in the public tenant, not in other one.

## Useful informations

How I get the error:
I creating my supertokens instance. I host my own core, using postgres and core v7 (image registry.supertokens.io/supertokens/supertokens-postgresql:7.0)
The backend uses the nodejs sdk.

Steps to reproduce:
- create two different database
- start db and core instance (db connection data passed with env `POSTGRESQL_CONNECTION_URI`
- setup backend in nodejs
   - setup a developer licence key
   - activated multi tenancy and passwordless login on public tenant
   - create new tenant (let's call it customTenant) passing new connection string (db dedicated to new tenant), activating only passwordless recipe
   - create a user1 in public tenant, and user2 in customTenant
   - invoke [this function](https://supertokens.com/docs/nodejs/modules/index.html#getUser) with user2.id, no user returned
   - invoke [this function](https://supertokens.com/docs/nodejs/modules/index.html#getUser) with user1.id, the correct user is returned

it seems that the function works only for public tenant.

On my use case, I'd like to pass also tenant id to the function.

Thanks
Eduardo

**Comment by sattvikc** [comment]
@edubacco the getUser function doesn't required a `tenantId`. It should be able to get any user within the app, across all tenants. When you say it seems to work only for public tenant, are you getting an `UNKNOWN_USER_ID` when you pass the userId of the user on customTenant?

**Comment by edubacco** [comment]
We miss a detail: the bug is related to the "update metadata" recipe.
A test to reproduce it:

```
import { getTestApp } from "../support/test_app"
import { Express } from "express"
import { expect } from "chai"
import { signInUp, updateUser } from "supertokens-node/recipe/passwordless"
import supertokens from "supertokens-node"
import { inspect } from "node:util"
import { faker } from "@faker-js/faker"
import { updateUserMetadata } from "supertokens-node/recipe/usermetadata"

describe.only("GET /admin/tenantId/users/userId", () => {
	let app: Express

	before(async () => {
		const testApp = await getTestApp()
		app = testApp.app
	})

	it("public ternant works", async () => {
		const publicUserEmail = "email_public@example.com"
		const publicUser = await signInUp({ tenantId: "public", email: publicUserEmail })

		const foundUser = await supertokens.getUser(publicUser.recipeUserId.getAsString())

		expect(foundUser?.emails).contain(publicUserEmail)
	})

	it.only("custom tenant works", async () => {
		const customUserEmail = "email_custom_3@example.com"
		const customUser = await signInUp({ tenantId: "ucsc", email: customUserEmail })
		const foundUser1 = await supertokens.getUser(customUser.recipeUserId.getAsString())

		expect(foundUser1?.emails).contain(customUserEmail)

		const responsePhone = await updateUser({
			recipeUserId: customUser.recipeUserId,
			phoneNumber: faker.phone.number().replace(/ /g, ""),
			email: customUserEmail,
		})

		const foundUser2 = await supertokens.getUser(customUser.recipeUserId.getAsString())

		expect(foundUser2?.emails).contain(customUserEmail)

		const responseMetadata = await updateUserMetadata(customUser.recipeUserId.getAsString(), {
			firstName: faker.person.firstName(),
			lastName: faker.person.lastName(),
		})

		const foundUser3 = await supertokens.getUser(customUser.recipeUserId.getAsString())
		console.log(inspect(foundUser3))
		expect(foundUser3?.emails).contain(customUserEmail)
	})
})
```

**Comment by edubacco** [comment]
The issue is reproducible also using the admin dashboard, when you try to edit metadata of a user you get an error and then you cannot access anymore the user

**Comment by rishabhpoddar** [comment]
This has been fixed in core version >= 8.0
